server/sonar-server/src/main/java/org/sonar/server/computation/metric/MetricRepositoryImpl.java

```diff
@@ -83,6 +84,11 @@ public Metric getById(long id) {
     return res;
   }
 
+  @Override
+  public Iterable<Metric> getAll() {
+    return FluentIterable.from(metricsByKey.values());
```

Comment:
Just using `FluentIterable.from` like that is either not necessary or not what you intended.

Either returns `metricsByKey.values()` directly because it is a `Set` which implements `Iterable` or use `toSet` or `ToList` of `FluentIterable` to create a copy of the `Set` and avoid having the internal map potentially modified externally

(go for the second option!)